src/index.js

```diff
@@ -0,0 +1,90 @@
+import updateTaskState from './taskState';
+import {
+  addTodo, editTodo, deleteTodo, clearFinishedTasks,
+} from './task';
+import { saveTodoList, getTodoList } from './functions';
+import './style.css';
+
+const todo = [];
+const todoListContainer = document.querySelector('.todo-list');
+const clearButton = document.querySelector('.clear');
+const enterButton = document.querySelector('.enter');
+const todoFormInput = document.querySelector('.todo-form-input');
+
+const displayTodoList = (todo) => {
+  todo.forEach((todoItem) => {
+    todoListContainer.innerHTML += `<li class='todo-item' data-id=${todoItem.index}>
+      <form class='flex'>
+        <input type='checkbox' class='todo-item-checkbox' ${todoItem.completed === true ? 'checked' : ''} name='todo' value='${todoItem.index}'>
+        <span type='text' contentEditable=true class='todo-item-input' data-id=${todoItem.index}>
+        ${todoItem.description}
+        </span>
+        <ion-icon class='icon' name="ellipsis-vertical-outline"></ion-icon>
+        <ion-icon class='icon delete hidden' name="trash-outline"></ion-icon>
+      </form>
+    </li>`;
+  });
+};
+
+todoFormInput.addEventListener('keydown', (e) => {
+  if (e.code === 'Enter') addTodo();
+});
+
+enterButton.addEventListener('click', () => {
+  addTodo();
+});
+
+if (!getTodoList()) {
+  saveTodoList(todo);
+}
+
+let savedTodoList = getTodoList();
+displayTodoList(savedTodoList);
+
+const checkboxes = document.querySelectorAll('input[name="todo"]');
+
+checkboxes.forEach((checkbox) => {
+  checkbox.addEventListener('change', () => {
+    savedTodoList = updateTaskState(getTodoList(), checkbox.value);
+    saveTodoList(savedTodoList);
+  });
+});
+
+// Input, Edit, Delete todo section
+const todoItemInputs = document.querySelectorAll('.todo-item-input');
+todoItemInputs.forEach((todoItemInput) => {
+  const ellipsis = todoItemInput.nextElementSibling;
+  const deleteIcon = todoItemInput.nextElementSibling.nextElementSibling;
+  const parentLiElement = todoItemInput.parentElement.parentElement;
+
+  todoItemInput.addEventListener('focus', () => {
+    ellipsis.style.display = 'none';
+    deleteIcon.style.display = 'block';
+    parentLiElement.style.backgroundColor = '#f7f4a8';
+  });
```

Comment:
Here you switch the icons when they focus on the text but I think is better for UX if this happens when you check the checkbox, what do you think?

Reply:
The idea is that users should be able to delete any task they want simply by clicking on that task. Having to mark a task as checked before you are allowed to delete such a task would actually be bad for UX.

Reply:
I got your point! yes even like in the video, I was thinking of deleting only when you complete a task, so if you checked is because is already completed.